src/uct/api/uct.h

```diff
@@ -1603,6 +1608,35 @@ UCT_INLINE_API ucs_status_t uct_ep_am_zcopy(uct_ep_h ep, uint8_t id, void *heade
     return ep->iface->ops.ep_am_zcopy(ep, id, header, header_length, iov, iovcnt, comp);
 }
 
+
+/**
+ * @ingroup UCT_AM
+ * @brief Register non-contiguous memory.
+ *
+ *
+ * @param [in]  ep           Destination endpoint handle.
+ * @param [in]  iov          Points to an array of @ref ::uct_iov_t structures.
+ *                           The @a iov pointer must be valid address of an array
+ *                           of @ref ::uct_iov_t structures. A particular structure
+ *                           pointer must be valid address. NULL terminated pointer
+ *                           is not required.
+ * @param [in]  iovcnt       Size of the @a iov data @ref ::uct_iov_t structures
+ *                           array. If @a iovcnt is zero, the data is considered empty.
+ *                           @a iovcnt is limited by @ref uct_iface_attr_cap_am_max_iov
```

Comment:
uct_iface_attr_cap_am_max_iov seems inapplicable - need to introduce mem_reg.max_iov?